rudder-core/src/main/scala/com/normation/rudder/services/servers/RemoveNodeService.scala

```diff
@@ -41,10 +47,10 @@ trait RemoveNodeService {
    * - clean the ou=Nodes
    * - clean the groups
    */
-  def removeNode(nodeId : NodeId, modId: ModificationId, actor:EventActor) : Box[Seq[LDIFChangeRecord]]
+  type DeletionResult = Box[Either[HookReturnCode.Stop,Either[HookReturnCode.Stop,Seq[LDIFChangeRecord]]]]
```

Comment:
I'm pretty sure we don't want that. 
The cases may be: 
- we have an error on deletion, because of pre-hooks or I/O
- we have a special pre-hook error code 100 that we want to display specificaly to the user, 
- we have a success and potentially a warning. 

I would model that with `Box[UserFeedback]`, where:
```
sealed trait UserFeedback
final object UserFeedback { 
  final case class DeletionForbiden(msg: String) extends UserFeedback
  final case class Success(msg: String, postHookResult: Box[HookResultCode]) extends UserFeedback 
}
```
And with that, you should never need to have the same kind of error processing in the UI pattern matching (generally a code smell). 
What do you think?